ecs-agent/acs/session/session_test.go

```diff
@@ -0,0 +1,1329 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package session
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"runtime"
+	"runtime/pprof"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	acsclient "github.com/aws/amazon-ecs-agent/ecs-agent/acs/client"
+	mock_session "github.com/aws/amazon-ecs-agent/ecs-agent/acs/session/mocks"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/session/testconst"
+	mock_api "github.com/aws/amazon-ecs-agent/ecs-agent/api/mocks"
+	rolecredentials "github.com/aws/amazon-ecs-agent/ecs-agent/credentials"
+	mock_credentials "github.com/aws/amazon-ecs-agent/ecs-agent/credentials/mocks"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/doctor"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/eventstream"
+	metricsfactory "github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/utils/retry"
+	mock_retry "github.com/aws/amazon-ecs-agent/ecs-agent/utils/retry/mock"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
+	mock_wsclient "github.com/aws/amazon-ecs-agent/ecs-agent/wsclient/mock"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/golang/mock/gomock"
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
```

Comment:
is it possible to leave a comment explaining why we should use string primitives instead of ecsacs message objects?

Reply:
Will take up this non-blocking comment in a followup PR. Thanks!